src/serac/physics/utilities/variational_form/integral.hpp

```diff
@@ -8,44 +19,64 @@
 #include "serac/physics/utilities/variational_form/finite_element.hpp"
 #include "serac/physics/utilities/variational_form/tuple_arithmetic.hpp"
 
+namespace serac {
+
 namespace impl {
 
+/**
+ * @brief Wrapper for mfem::Reshape compatible with Serac's finite element space types
+ * @tparam space A test or trial space
+ * @param[in] u The raw data to reshape into an mfem::DeviceTensor
+ * @param[in] n1 The first dimension to reshape into
+ * @param[in] n2 The second dimension to reshape into
+ * @see mfem::Reshape
+ */
 template <typename space>
 auto Reshape(double* u, int n1, int n2)
 {
+  // TODO: Helpful static_assert
```

Comment:
what should test here with a `static_assert`?

Reply:
Was thinking that we'd implement a `is_finite_element_space` trait and then `static_assert(detail::is_finite_element_space<space>::value);` or something - there's a fair number of places where I thought we could go back later and add these little helpers that get us slightly less incomprehensible template error messages from the compiler, but I don't think they're critical for this big initial merge

Reply:
I 100% agree with the spirit of using `static_assert` to spare the user nasty template compiler errors, but I think `impl::Reshape` might be the wrong place to do it, since it's so deep into the implementation. Instead, we could catch the same issue higher up by `static_assert`ing on `is_finite_element_space<test_space>::value` at the WeakForm or Integral level, and it would output a smaller template error "backtrace".

Reply:
That definitely makes more sense, it's also possible that I've already left a similar FIXME/TODO comment at a higher level that would obviate the need for this one